tests/mut-ref-functionalization/src/lib.rs

```diff
@@ -6,6 +6,11 @@ fn index_mutation(x: core::ops::Range<usize>, a: &'static [u8]) {
     v[1] = 3;
 }
 
+fn index_mutation_unsize(mut x: [u8; 12]) -> u8 {
+    x[4..5].copy_from_slice(&[1, 2]);
+    42
+}
+
```

Comment:
Can we add a test like this?
```rust
struct S {
    b: [u8; 5],
}

impl S {
    fn update(&mut self, x: u8) {
        self.b[0] = x;
    }
}
```
I think the output here doesn't look quite right.
```ocaml
let self:t_S = { self with f_b = Rust_primitives.Hax.update_at (f_b self <: t_S)
```

Reply:
Ah, I thought this was fixed.. Though I think I know what the problem is, will push soon